src/common/identity_parser/identity_parser.cpp

```diff
@@ -37,6 +39,39 @@ kvp::ExpectedKeyValuesMap GetIdentityData(const string &identity_data_generator)
 	return ex_key_values;
 }
 
+string DumpIdentityData(kvp::KeyValuesMap identity_data) {
+	stringstream top_ss;
+	top_ss << "{";
+	auto key_vector = common::GetMapKeyVector(identity_data);
+	std::sort(key_vector.begin(), key_vector.end());
+	for (const auto &key : key_vector) {
+		top_ss << json::EscapeString(key);
+		top_ss << R"(:)";
+		if (identity_data[key].size() == 1) {
+			top_ss << "\"" + json::EscapeString(identity_data[key][0]) + "\"";
+		} else {
+			stringstream items_ss;
+			items_ss << "[";
+			for (const auto &str : identity_data[key]) {
+				items_ss << "\"" + json::EscapeString(str) + "\",";
+			}
+			auto items_str = items_ss.str();
+			// replace the trailing comma with the closing square bracket
+			items_str[items_str.size() - 1] = ']';
```

Comment:
I think this will break if `identity_data` is empty. The question is, can it be? You should probably use the same check as below.

Reply:
On empty `identity_data` the code won't get into this for loop. Then, it could be a problem on a map entry with a key and a `vector<string>` of 0 elements.

For the later, I added some unit tests for this and figured that it cannot really happen because even on blank values (say, an script echoing `key="`) the parser inserts an empty string in the vector.

I think it is okey to leave as is (with the tests, coming up in the next push).

For reference, if we fix here we must also follow-up [here](https://github.com/mendersoftware/mender/blob/2ad0d486f5aba4f760a3af85b00321ae85f5c446/src/mender-update/inventory.cpp#L104).